README.md

```diff
@@ -0,0 +1,234 @@
+[![Maintainability](https://api.codeclimate.com/v1/badges/22cfed180fd6032fe29b/maintainability)](https://codeclimate.com/github/opusonesolutions/carsons/maintainability)
+
+This is an implementation of Carson's Equations, a mathematical model
+for deriving the equivalent impedance of an AC transmission or
+distribution line.
+
+Implementation
+--------------
+
+`carsons` is developed using python 3.6 support for unicode characters
+like π, ƒ, ρ, μ, ω etc. This feature allows us to avoid translating the
+problem into a more typical programming syntax, so the code is dense and
+can easily be compared to published formulations of the problem.
+
+For example, we implement the kron reduction, a matrix decomposition
+step, using unicode notation to indicate the slightly different meaning
+of impedance values before and after a kron reduction:
+
+```python
+def perform_kron_reduction(z_primitive):
+     Ẑpp, Ẑpn = z_primitive[0:3, 0:3], z_primitive[0:3, 3:]
+     Ẑnp, Ẑnn = z_primitive[3:,  0:3], z_primitive[3:,  3:]
+     Z_abc = Ẑpp - Ẑpn @ inv(Ẑnn) @ Ẑnp
+     return Z_abc
+```
+
+Take a look at the [source
+code](https://github.com/opusonesolutions/carsons/blob/add-documentation/carsons/carsons.py)
+to see more cool unicode tricks!
+
+Installation
+------------
+
+```bash
+~/$ pip install carsons
+```
+
+Usage
+-----
+
+Carsons model requires a line model object that maps each phase to
+properties of the conductor for that phase.
+
+```python
+from carsons import CarsonsEquations, calculate_impedance
+
+class Line:
+   geometric_mean_radius: {
+       'A': geometric_mean_radius_A
+       ...
+   }
+   resistance: {
+        'A': per-length resistance of conductor A in ohms
+        ...
+   }
+   wire_positions: {
+        'A': (x, y) cross-sectional position of the conductor in meters
+        ...
+   }
+   phases: {'A', ... }
+     # map of phases 'A', 'B', 'C' and 'N<>' which are described in the
+     # gmr, r and phase_positions attributes
+
+line_impedance = calculate_impedance(CarsonsEquations(Line()))
+```
+
+The model supports any combination of ABC phasings (for example BC, BCN
+etc...) including systems with multiple neutral cables; any phases that
+are not present in the model will have zeros in the columns and rows
+corresponding to that phase.
+
+Multiple neutrals are supported, as long as they have unique labels
+starting with `N` (e.g. `Neutral1`, `Neutral2`).
+
+Intermediate results such as primitive impedance matrix are also
+available.
+
+```python
+z_primitive = CarsonsEquations(Line()).build_z_primitive()
+```
+
+For examples of how to use the model, see the [overhead wire
+tests](https://github.com/opusonesolutions/carsons/blob/master/tests/test_carsons.py).
+
+`carsons` is tested against several cable configurations from the [IEEE
+4-bus test network](http://sites.ieee.org/pes-testfeeders/resources/).
```

Comment:
Should this just be IEEE networks? Should we also mention that we tested against OpenDSS as well?

Reply:
Good idea. I made it just reference 'ieee test feeders' and added a note about opendss documentation as well as a link in the references